sdk/core/Azure.Core/src/ResourceIdentifier.cs

```diff
@@ -180,22 +184,22 @@ private static ResourceIdentifier AppendNext(ResourceIdentifier parent, string[]
         /// <summary>
         /// The resource type of the resource.
         /// </summary>
-        public ResourceType ResourceType { get; private set; }
+        public ResourceType ResourceType { get; }
 
         /// <summary>
         /// The name of the resource.
         /// </summary>
-        public string Name { get; private set; }
+        public string Name { get; }
 
         /// <summary>
         /// The immediate parent containing this resource.
         /// </summary>
-        public ResourceIdentifier Parent { get; private set; }
+        public ResourceIdentifier? Parent { get; private set; }
```

Comment:
Isn't `Parent` always set?  It looks like if the caller passed null, it gets set to `Root`... what is the intended functionality from the perspective of the business logic - is it acceptable to have a null parent?

Reply:
It can definitely be null.  In the example we are using for perf

```
ResourceIdentifier id = new ResourceIdentifier("/subscriptions/0c2f6471-1bf0-4dda-aec3-cb9272f09575/resourceGroups/myRg/providers/Microsoft.Compute/virtualMachines/myVm/subnets/mysubnet"); //VirtualMachine id
id = id.Parent; //ResourceGroup id
id = id.Parent; //Subscription id
id = id.Parent; //Tenant id
id = id.Parent; //null
```